From f0d1d0b290a543a98c04308bc3a7453471009871 Mon Sep 17 00:00:00 2001
From: Markus Armbruster <armbru@redhat.com>
Date: Wed, 31 Mar 2010 13:25:08 -0300
Subject: [PATCH 53/66] Revert "qdev: Use QError for 'device not found' error"

RH-Author: Markus Armbruster <armbru@redhat.com>
Message-id: <1270041921-28969-54-git-send-email-armbru@redhat.com>
Patchwork-id: 8223
O-Subject: [PATCH 53/66] Revert "qdev: Use QError for 'device not found' error"
Bugzilla: 579470
RH-Acked-by: Kevin Wolf <kwolf@redhat.com>
RH-Acked-by: Juan Quintela <quintela@redhat.com>
RH-Acked-by: Luiz Capitulino <lcapitulino@redhat.com>

This reverts commit 3ced9f7a36189aed94d8bf86f3f5087a53012455.

The next commit will convert all of qdev_device_add() to QError, and
it'll be clearer with this partial conversion reverted.
(cherry picked from commit 3801cf8ae58726b89a37f4594b7c9095e1050501)
---
 hw/qdev.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

Signed-off-by: Eduardo Habkost <ehabkost@redhat.com>
---
 hw/qdev.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/hw/qdev.c b/hw/qdev.c
index 27dc8df..43dff04 100644
--- a/hw/qdev.c
+++ b/hw/qdev.c
@@ -29,7 +29,6 @@
 #include "qdev.h"
 #include "sysemu.h"
 #include "monitor.h"
-#include "qerror.h"
 
 static int qdev_hotplug = 0;
 
@@ -208,7 +207,8 @@ DeviceState *qdev_device_add(QemuOpts *opts)
     /* find driver */
     info = qdev_find_info(NULL, driver);
     if (!info || info->no_user) {
-        qerror_report(QERR_DEVICE_NOT_FOUND, driver);
+        error_report("Device \"%s\" not found.  Try -device '?' for a list.",
+                     driver);
         return NULL;
     }
 
-- 
1.7.0.3

